src/Compilers/Core/MSBuildTaskTests/TestUtilities/IntegrationTestBase.cs

```diff
@@ -0,0 +1,174 @@
+
+namespace Microsoft.CodeAnalysis.BuildTasks.UnitTests;
+
+public abstract class IntegrationTestBase : TestBase
+{
+    protected static readonly string? s_msbuildDirectory;
+
+    static IntegrationTestBase()
+    {
+        s_msbuildDirectory = DesktopTestHelpers.GetMSBuildDirectory();
+    }
+
+    protected readonly ITestOutputHelper _output;
+    protected readonly string? _msbuildExecutable;
+    protected readonly TempDirectory _tempDirectory;
+    protected string _buildTaskDll;
+
+    protected IntegrationTestBase(ITestOutputHelper output)
+    {
+        if (s_msbuildDirectory == null)
+        {
+            output.WriteLine("Could not locate MSBuild");
+        }
+
+        _output = output;
+        _msbuildExecutable = s_msbuildDirectory == null ? null : Path.Combine(s_msbuildDirectory, "MSBuild.exe");
+        _tempDirectory = Temp.CreateDirectory();
+        _buildTaskDll = typeof(ManagedCompiler).Assembly.Location;
+    }
+
+    private static IEnumerable<KeyValuePair<string, string>> AddForLoggingEnvironmentVars(IEnumerable<KeyValuePair<string, string>>? vars)
+    {
+        vars = vars ?? new KeyValuePair<string, string>[] { };
+        if (!vars.Where(kvp => kvp.Key == "RoslynCommandLineLogFile").Any())
+        {
+            var list = vars.ToList();
+            list.Add(new KeyValuePair<string, string>(
+                "RoslynCommandLineLogFile",
+                typeof(IntegrationTestBase).Assembly.Location + ".client-server.log"));
+            return list;
+        }
+        return vars;
+    }
+
+    protected static ProcessResult RunCommandLineCompiler(
+        string compilerPath,
+        string arguments,
+        string currentDirectory,
+        IEnumerable<KeyValuePair<string, string>>? additionalEnvironmentVars = null)
+    {
+        return ProcessUtilities.Run(
+            compilerPath,
+            arguments,
+            currentDirectory,
+            additionalEnvironmentVars: AddForLoggingEnvironmentVars(additionalEnvironmentVars));
+    }
+
+    protected static ProcessResult RunCommandLineCompiler(
+        string compilerPath,
+        string arguments,
+        TempDirectory currentDirectory,
+        IEnumerable<KeyValuePair<string, string>> filesInDirectory,
+        IEnumerable<KeyValuePair<string, string>>? additionalEnvironmentVars = null)
+    {
+        foreach (var pair in filesInDirectory)
+        {
+            TempFile file = currentDirectory.CreateFile(pair.Key);
+            file.WriteAllText(pair.Value);
+        }
+
+        return RunCommandLineCompiler(
+            compilerPath,
+            arguments,
+            currentDirectory.Path,
+            additionalEnvironmentVars: AddForLoggingEnvironmentVars(additionalEnvironmentVars));
+    }
+
+    [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/79907")]
+    public void StdLib_Csc(bool useSharedCompilation, bool disableSdkPath)
+    {
+        if (_msbuildExecutable == null) return;
```

Comment:
Would it be helpful to log that we are essentially skipping the test in this scenario and in similar place in `StdLib_Vbc` test below?

Reply:
We log that in the constructor which runs for each test separately, so the test's log contain the information:

https://github.com/dotnet/roslyn/blob/e927f0bd5a833a78eaf032f673d2f36b6e0ed070/src/Compilers/Core/MSBuildTaskTests/TestUtilities/IntegrationTestBase.cs#L32-L37